### PR TITLE
Reverses the order of layer resolution 

### PR DESCRIPTION
**Pull request by harryi3t**
while performing  `variableScope#toObject`

This was a bug since `variableScope#get` gives the first argument more precedence

### CODE REVIEW

File: lib/collection/variable-scope.js [hunk 1]
```diff
@@ -146,10 +146,9 @@ _.assign(VariableScope.prototype, /** @lends VariableScope.prototype */ {
         }
 
         var mergedLayers = {},
-            i,
-            ii;
+            i;
 
-        for (i = 0, ii = this._layers.length; i < ii; i++) {
+        for (i = this._layers.length - 1; i >= 0; i--) {
```
Comment (shamasis) [inline lib/collection/variable-scope.js]: Infinity bug

**Comment by harryi3t** [comment]
checklist for the skipped test
- [ ] Should skip on travis/pr v4.x
- [ ] Should skip on travis/branch v4.x
- [ ] Should not skip on travis/branch v6.x
- [ ] Should not skip on travis/pr v6.x
- [ ] Should not skip on appveyor/branch v6.x
- [ ] Should not skip on appveyor/branch v4.x
- [ ] Should not skip on appveyor/pr v6.x
- [ ] Should not skip on appveyor/pr v4.x

File: test/unit/url.test.js [hunk 2]
```diff
@@ -806,7 +811,7 @@ describe('Url', function () {
     });
 
     describe('Security', function () {
-        describe('ReDOS', function () {
+        travisNodeV4Skip('ReDoS', function () {
```
Comment (kamalaknn) [inline test/unit/url.test.js]: Need to add a comment here explaining why it was skipped

Comment (kunagpal) [reply test/unit/url.test.js]: Also, this isn't going to be used anywhere else, so you can use a literal expression on line 814.

**Comment by harryi3t** [comment]
@kamalaknn @kunagpal Updated the PR :)

**Pull request merged**